src/wayland/data_device/mod.rs

```diff
@@ -114,7 +115,7 @@ pub trait DataDeviceHandler: Sized + ClientDndGrabHandler + ServerDndGrabHandler
     ///
     /// * `mime_type` - the requested mime type
     /// * `fd` - the fd to write into
-    fn send_selection(&mut self, mime_type: String, fd: RawFd) {}
+    fn send_selection(&mut self, mime_type: String, fd: OwnedFd) {}
```

Comment:
@vberger is this correct to just close the fd by dropping it here

Reply:
Looks right, the function passes FD ownership to the handler, which then decides what to do with it. :+1: